### PR TITLE
[9.0][ADD] account_invoice_fixed_discount

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Attend issue https://github.com/OCA/account-invoicing/issues/310

Account Invoice Fixed Discount
======================

This module extends the functionality of Invoicing to allow you to apply fixed
amount discounts at invoice line level.

The module also extends the invoice report to show fixed discount.

### CODE REVIEW

**Review by JordiBForgeFlow — APPROVED** [review]
Great one!

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Tested on runbot

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :+1:

File: account_invoice_fixed_discount/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def get_taxes_values(self):
+        self.ensure_one()
+        vals = {}
+        for line in self.invoice_line_ids.filtered(lambda l: l.discount_fixed):
```
Comment (moylop260) [inline account_invoice_fixed_discount/models/account_invoice.py]: filtered('discount_fixed')

**Review by moylop260 — CHANGES_REQUESTED** [review]
Review the beta checks please
https://travis-ci.org/OCA/account-invoicing/jobs/271739620#L743

**Comment by lreficent** [comment]
@moylop260 the method `_compute_price` is api.one in standard odoo: https://github.com/odoo/odoo/blob/9.0/addons/account/models/account_invoice.py#L1103. The other changes has been made.

**Comment by LoisRForgeFlow** [comment]
@moylop260 the method _compute_price is api.one in standard odoo: https://github.com/odoo/odoo/blob/9.0/addons/account/models/account_invoice.py#L1103. The other changes have been made.

File: account_invoice_fixed_discount/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            if line.discount and line.discount_fixed:
+                raise ValidationError(
+                    _("You can only set one type of discount per line."))
+
+    @api.one
+    @api.depends('price_unit', 'discount', 'invoice_line_tax_ids', 'quantity',
+                 'product_id', 'invoice_id.partner_id',
+                 'invoice_id.currency_id', 'invoice_id.company_id',
+                 'discount_fixed')
+    def _compute_price(self):
```
Comment (moylop260) [inline account_invoice_fixed_discount/models/account_invoice.py:64]: @pedrobaeza @lasley 
What do you think about a inherit of a computed method?

Comment (pedrobaeza) [reply account_invoice_fixed_discount/models/account_invoice.py:64]: Well, we do something similar in https://github.com/OCA/account-invoicing/blob/7a2aa5c6ed53cb0e9d853219b04e7224a93bd2ee/account_invoice_triple_discount/models/account_invoice.py#L49, and these cases inheritance will fail, but there's no other way to do it. You can mention the incompatibility in the README.

Comment (LoisRForgeFlow) [reply account_invoice_fixed_discount/models/account_invoice.py:64]: what do you mean? is it impossible to have installed and working fine both this module and yours?

Comment (pedrobaeza) [reply account_invoice_fixed_discount/models/account_invoice.py:64]: Yeah, as both modifies the price_unit as trick for having the rest without taking into account possible previous modifications. You can't have a fixed price discount and also second and third discount. If you see we can find a way to conciliate both....

Comment (LoisRForgeFlow) [reply account_invoice_fixed_discount/models/account_invoice.py:64]: We could try to add a hook module which we both can inherit into our modules. Do you think that would be possible?

**Comment by LoisRForgeFlow** [comment]
@pedrobaeza @moylop260 I added a warning in the README about the incompatibility with account_invoice_triple_discount. Can we move forward?

**Review by moylop260 — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
Rebased and squashed.

**Pull request merged**